src/main/java/com/synopsys/integration/alert/channel/jira/JiraIssueHandler.java

```diff
@@ -124,43 +125,54 @@ public String createOrUpdateIssues(final FieldAccessor fieldAccessor, final Mess
-                    final String issueKey = issue.getKey();
-                    addIssueProperties(issueKey, providerName, topic, subTopic, arbitraryItem, trackingKey);
-                    addComment(issueKey, "This issue was automatically created by Alert.");
+                final Optional<IssueComponent> existingIssueComponent = retrieveExistingIssue(providerName, topic, subTopic, arbitraryItem, trackingKey);
+                if (existingIssueComponent.isPresent()) {
+                    final IssueComponent issueComponent = existingIssueComponent.get();
+                    final String transitionKey = (ItemOperation.DELETE.equals(operation)) ? JiraDescriptor.KEY_RESOLVE_WORKFLOW_TRANSITION : JiraDescriptor.KEY_OPEN_WORKFLOW_TRANSITION;
+                    final String issueKey = transitionIssue(issueComponent.getKey(), fieldAccessor, transitionKey);
                     issueKeys.add(issueKey);
+                    if (commentOnIssue) {
+                        String operationComment = createOperationComment(operation, arbitraryItem.getCategory(), providerName, combinedItems);
+                        addComment(issueKey, operationComment);
+                    }
                 } else {
-                    logger.warn("Expected to find an existing issue with key '{}' but none existed.", trackingKey);
+                    if (ItemOperation.ADD.equals(operation) || ItemOperation.UPDATE.equals(operation)) {
+                        fieldsBuilder.setProject(projectId);
+                        fieldsBuilder.setIssueType(issueType);
+                        final String username = fieldAccessor.getString(JiraDescriptor.KEY_JIRA_ADMIN_EMAIL_ADDRESS).orElseThrow(() -> new AlertException("Expected to be passed a jira user email address."));
+                        final IssueResponseModel issue = issueService.createIssue(new IssueCreationRequestModel(username, issueType, projectName, fieldsBuilder, List.of()));
+                        if (issue == null || StringUtils.isBlank(issue.getKey())) {
+                            throw new AlertException("There was an problem when creating this issue.");
+                        }
+                        final String issueKey = issue.getKey();
+                        addIssueProperties(issueKey, providerName, topic, subTopic, arbitraryItem, trackingKey);
+                        addComment(issueKey, "This issue was automatically created by Alert.");
+                        issueKeys.add(issueKey);
+                    } else {
+                        logger.warn("Expected to find an existing issue with key '{}' but none existed.", trackingKey);
+                    }
                 }
+            } catch (JiraMissingTransitionException e) {
+                missingTransitions.add(e.getTransition());
             }
         }
+        if (!missingTransitions.isEmpty()) {
+            final String joinedMissingTransitions = missingTransitions.stream().collect(Collectors.joining(","));
```

Comment:
On a UI related note, should we join with `", "` rather than `","` (add a space)?